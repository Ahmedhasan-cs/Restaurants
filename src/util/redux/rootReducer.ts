import {combineReducers} from 'redux';
import authReducer from './slice/authenticationSlice';
import cuisinesReducer from './slice/cuisineSlice';

const rootReducer = combineReducers({
  auth: authReducer,
  cuisines: cuisinesReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
