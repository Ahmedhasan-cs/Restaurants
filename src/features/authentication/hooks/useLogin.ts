import {useState} from 'react';
import { useReduxDispatch, useReduxSelector } from '@util/redux/store';
import { hasError, inLoading, loginRequest } from '@util/redux/slice/authenticationSlice';

const emailRegex = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;

interface UseLoginReturn {
  email: string;
  setEmail: (email: string) => void;
  password: string;
  setPassword: (password: string) => void;
  isEmailValid: boolean;
  isPasswordValid: boolean;
  isFormSubmitted: boolean;
  submit: () => void;
  isLoading: boolean;
  error: string | null;
}

export const useLogin = (): Readonly<UseLoginReturn> => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [isFormSubmitted, setIsFormSubmitted] = useState(false);
  const dispatch = useReduxDispatch();
  const isEmailValid = emailRegex.test(email);
  const isPasswordValid = password.length >= 6;

  const isLoading = useReduxSelector(inLoading);
  const error = useReduxSelector(hasError);

  function submit() {
    setIsFormSubmitted(true);

    if (!isEmailValid || !isPasswordValid) return;

    handleLogin();
  }

  function handleLogin() {
    try {
      dispatch(loginRequest({ email, password }));
    } catch (err) {
      console.log("Login failed. Please try again.");
    } finally {
    }
  };

  return {
    email,
    setEmail,
    password,
    setPassword,
    isEmailValid,
    isPasswordValid,
    isFormSubmitted,
    submit,
    isLoading,
    error
  };
};