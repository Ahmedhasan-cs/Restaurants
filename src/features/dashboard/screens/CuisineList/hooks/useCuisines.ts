import {useCallback, useEffect, useMemo, useState} from 'react';
import { useReduxDispatch, useReduxSelector } from '@util/redux/store';
import { cuisinesData, fetchCuisinesStart, hasError, inLoading, setSelectedCuisine } from '@util/redux/slice/cuisineSlice';
import { Cuisine, CuisineCount, Cuisines } from '@util/api/models/CuisineResponse';

interface CuisinesOperators {
  cuisines: Cuisines;
  isLoading: boolean;
  error: string | null;
  fetchCuisines: () => void;
  cuisineCounts: CuisineCount[];
  handleSelectedCuisine : (cuisine: Cuisine) => void;
}

export const useCuisines = (): Readonly<CuisinesOperators>  => {
  const dispatch = useReduxDispatch();
  const cuisines = useReduxSelector(cuisinesData);
  const isLoading = useReduxSelector(inLoading);
  const error = useReduxSelector(hasError);
  
  const cuisineCounts: CuisineCount[] = useMemo(() => {
    if (!cuisines || Object.keys(cuisines).length === 0) {
      console.warn("Cuisines object is empty or undefined");
      return [];
    }
  
    return Object.entries(cuisines).map(([title, status]) => ({
      title: title as Cuisine,
      count: (status.open?.length || 0) + (status.close?.length || 0),
    }));
  }, [cuisines]);

  const handleSelectedCuisine = useCallback((cuisine: Cuisine) => {
    dispatch(setSelectedCuisine(cuisine));
  }, [dispatch]);


  const fetchCuisines = useCallback(() => {
    dispatch(fetchCuisinesStart());
  }, [dispatch]);

  useEffect(() => {
      fetchCuisines();
    }, [fetchCuisines]);


  return {
    cuisines,
    isLoading,
    error,
    fetchCuisines,
    cuisineCounts,
    handleSelectedCuisine
  };
};
