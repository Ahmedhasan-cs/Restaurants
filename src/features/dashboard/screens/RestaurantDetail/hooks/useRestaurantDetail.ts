import { useReduxSelector } from '@util/redux/store';
import { hasError, inLoading, selectedRestaurantData } from '@util/redux/slice/cuisineSlice';
import { Restaurant } from '@util/api/models/CuisineResponse';

interface ResturantsOperators {
  isLoading: boolean;
  error: string | null;
  selectedRestaurant: Restaurant;
}

export const useRestaurantDetail = (): Readonly<ResturantsOperators>  => {
  const selectedRestaurant = useReduxSelector(selectedRestaurantData);
  const isLoading = useReduxSelector(inLoading);
  const error = useReduxSelector(hasError);
  
  return {
    isLoading,
    error,
    selectedRestaurant,
  };
};
