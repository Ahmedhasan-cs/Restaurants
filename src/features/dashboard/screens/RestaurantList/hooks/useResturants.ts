import {useCallback, useMemo} from 'react';
import { useReduxDispatch, useReduxSelector } from '@util/redux/store';
import { cuisinesData, hasError, inLoading, selectedCuisineData, setSelectedRestaurant } from '@util/redux/slice/cuisineSlice';
import { Cuisine, Restaurant } from '@util/api/models/CuisineResponse';

interface ResturantsOperators {
  isLoading: boolean;
  error: string | null;
  selectedCuisine: Cuisine;
  resturants: Restaurant[];
  handleSelectedRestaurant : (resturant: Restaurant) => void;
}

export const useResturants = (): Readonly<ResturantsOperators>  => {
  const dispatch = useReduxDispatch();
  const cuisines = useReduxSelector(cuisinesData);
  const selectedCuisine = useReduxSelector(selectedCuisineData);
  const isLoading = useReduxSelector(inLoading);
  const error = useReduxSelector(hasError);
  
  const resturants: Restaurant[] = useMemo(() => {
    if (!cuisines || Object.keys(cuisines).length === 0) {
      console.warn("Cuisines object is empty or undefined");
      return [];
    }
  
    const selectedCuisineResturants = cuisines[selectedCuisine];
    if (!selectedCuisineResturants) {
      console.warn(`No resturants found for ${selectedCuisine}`);
      return [];
    } 
    const updatedRestaurants = [
      ...selectedCuisineResturants.open.map(rest => ({
        ...rest,
        isOpened: true
      })),
      ...selectedCuisineResturants.close.map(rest => ({
        ...rest,
        isOpened: false
      }))
    ];
    return updatedRestaurants
  }, [cuisines]);

  const handleSelectedRestaurant = useCallback((resturant: Restaurant) => {
    dispatch(setSelectedRestaurant(resturant));
  }, [dispatch]);

  return {
    isLoading,
    error,
    selectedCuisine,
    resturants,
    handleSelectedRestaurant
  };
};
