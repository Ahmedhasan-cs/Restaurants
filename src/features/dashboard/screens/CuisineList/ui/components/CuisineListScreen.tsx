import {styles} from '../styles/cuisineListStyle';
import {View, FlatList, Alert} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import CuisineItem from './CuisineItem';
import {
  CuisinesNavigationProps,
  RouteNames,
} from '@navigators/dashboard/DashboardNavigationProps';
import {useCuisines} from '@features/dashboard/screens/CuisineList/hooks/useCuisines';
import {Cuisine} from '@util/api/models/CuisineResponse';
import {useEffect} from 'react';

const CuisineListScreen = ({route, navigation}: CuisinesNavigationProps) => {
  const {error, cuisineCounts, handleSelectedCuisine} = useCuisines();

  const handlePress = (cuisine: Cuisine): void => {
    console.log('Cuisine pressed', cuisine);
    handleSelectedCuisine(cuisine);
    navigation.navigate(RouteNames.Restaurants);
  };

  useEffect(() => {
    if (error) {
      Alert.alert('Error', error);
    }
  }, [error]);

  return (
    <SafeAreaView style={styles.flexOne}>
      <View style={styles.container}>
        <FlatList
          data={cuisineCounts}
          keyExtractor={item => item.title}
          renderItem={({item}) => (
            <CuisineItem cuisine={item} handlePress={handlePress} />
          )}
          contentContainerStyle={styles.list}
        />
      </View>
    </SafeAreaView>
  );
};

export default CuisineListScreen;
