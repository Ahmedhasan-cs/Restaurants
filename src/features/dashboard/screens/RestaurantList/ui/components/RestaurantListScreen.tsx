import {styles} from '../styles/restaurantListStyle';
import {View, FlatList, Alert} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import {
  RestaurantsNavigationProps,
  RouteNames,
} from '@navigators/dashboard/DashboardNavigationProps';
import {Restaurant} from '@util/api/models/CuisineResponse';
import {useResturants} from '../../hooks/useResturants';
import RestaurantItem from './RestaurantItem';
import HeaderView from '@components/HeaderView';
import {useCallback, useEffect} from 'react';

const RestaurantListScreen = ({
  route,
  navigation,
}: RestaurantsNavigationProps) => {
  const {
    selectedCuisine,
    error,
    isLoading,
    resturants,
    handleSelectedRestaurant,
  } = useResturants();

  useEffect(() => {
    if (error) {
      Alert.alert('Error', error);
    }
  }, [error]);

  const handlePress = useCallback((resturant: Restaurant) => {
    handleSelectedRestaurant(resturant);
    navigation.navigate(RouteNames.Detail);
  }, []);

  const handleBack = useCallback(() => {
    navigation.goBack();
  }, []);

  return (
    <SafeAreaView style={styles.flexOne}>
      <View style={styles.container}>
        <HeaderView title={selectedCuisine} backAction={handleBack} />
        <FlatList
          data={resturants}
          keyExtractor={item => item.id}
          renderItem={({item}) => (
            <RestaurantItem
              restaurant={item}
              handlePress={handlePress}
              testID="cuisine-item"
            />
          )}
          contentContainerStyle={styles.list}
        />
      </View>
    </SafeAreaView>
  );
};

export default RestaurantListScreen;
