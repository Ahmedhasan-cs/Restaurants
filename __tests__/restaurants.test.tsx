import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react-native';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import {
  CuisinesNavigationProps,
  RestaurantsNavigationProps,
  RouteNames,
} from '@navigators/dashboard/DashboardNavigationProps';
import CuisineListScreen from '@features/dashboard/screens/CuisineList/ui/components/CuisineListScreen';
import cuisinesReducer from '@util/redux/slice/cuisineSlice';
import rootReducer from '@util/redux/rootReducer';
import reducer from '@util/redux/slice/cuisineSlice';
import {cuisinesData} from '@util/api/mocks/cuisines';
import createSagaMiddleware from 'redux-saga';
import RestaurantListScreen from '@features/dashboard/screens/RestaurantList/ui/components/RestaurantListScreen';
import {useResturants} from '@features/dashboard/screens/RestaurantList/hooks/useResturants';

// Mock navigation
const mockNavigate = jest.fn();
const mockRoute = {params: {}};
const mockNavigation: Partial<RestaurantsNavigationProps['navigation']> = {
  navigate: mockNavigate,
};

jest.mock(
  '@features/dashboard/screens/RestaurantList/hooks/useResturants',
  () => ({
    useResturants: jest.fn(),
  }),
);

export const createTestStore = (customState = {}) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState: {
      cuisines: {
        cuisines: cuisinesData,
        loading: false,
        error: null,
      },
      auth: {
        isAuthorized: true,
        loading: false,
        error: null,
      },
    },
    middleware: getDefaultMiddleware => getDefaultMiddleware({thunk: false}), // No sagas in tests
  });
};

export const renderWithProviders = (
  ui,
  {store} = {store: createTestStore()},
) => {
  return render(<Provider store={store}>{ui}</Provider>);
};

describe('CuisineListScreen Tests', () => {
  beforeEach(() => {
    (useResturants as jest.Mock).mockReturnValue({
      selectedCuisine: 'italian',
      error: null,
      isLoading: false,
      resturants: [
        {
          id: '6109200ecdbea9077786439c',
          restaurantName: 'Lunchpod',
          shortDesc: 'Relax your belly',
          currency: 'EUR',
          deliveryCost: 2.15,
          rating: 6,
          minOrder: 10,
          deliveryTime: '20-25 min',
          speciality: 'DINNER',
          imageUrl:
            'https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_feedback002.jpg',
        },
        {
          id: '6109200e7363c5fce8c4262e',
          restaurantName: 'Magneato',
          shortDesc: 'Soul for food',
          currency: 'EUR',
          deliveryCost: 1.91,
          rating: 6,
          minOrder: 15,
          deliveryTime: '15-20 min',
          imageUrl:
            'https://www.top10berlin.de/wp-content/uploads/2023/11/umami_top10berlin003.jpg',
        },
      ],
      handleSelectedRestaurant: jest.fn(),
    });
  });

  it('should render cuisines from Redux store', () => {
    const {getByText, getAllByTestId} = renderWithProviders(
      <RestaurantListScreen
        navigation={mockNavigation as any}
        route={mockRoute as any}
      />,
    );

    expect(getByText('Italian')).toBeTruthy();
    expect(getByText('Lunchpod')).toBeTruthy();

    const items = getAllByTestId('cuisine-item');
    expect(items.length).toBe(2); // ✅ Check if 3 items are rendered
  });

  it('renders CuisineListScreen correctly and matches snapshot', () => {
    const tree = render(
      <RestaurantListScreen
        navigation={mockNavigation as any}
        route={mockRoute as any}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot(); // ✅ Snapshot comparison
  });
});
