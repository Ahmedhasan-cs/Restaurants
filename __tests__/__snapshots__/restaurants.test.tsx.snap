// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CuisineListScreen Tests renders CuisineListScreen correctly and matches snapshot 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#fff",
        "paddingTop": 20,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "width": 40,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "black",
                "fontSize": 20,
              },
              undefined,
              {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          î— 
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#333",
            "flex": 1,
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Italian
      </Text>
      <View
        style={
          {
            "width": 40,
          }
        }
      />
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "paddingHorizontal": 16,
        }
      }
      data={
        [
          {
            "currency": "EUR",
            "deliveryCost": 2.15,
            "deliveryTime": "20-25 min",
            "id": "6109200ecdbea9077786439c",
            "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_feedback002.jpg",
            "minOrder": 10,
            "rating": 6,
            "restaurantName": "Lunchpod",
            "shortDesc": "Relax your belly",
            "speciality": "DINNER",
          },
          {
            "currency": "EUR",
            "deliveryCost": 1.91,
            "deliveryTime": "15-20 min",
            "id": "6109200e7363c5fce8c4262e",
            "imageUrl": "https://www.top10berlin.de/wp-content/uploads/2023/11/umami_top10berlin003.jpg",
            "minOrder": 15,
            "rating": 6,
            "restaurantName": "Magneato",
            "shortDesc": "Soul for food",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="cuisine-item"
          >
            <View
              style={
                {
                  "backgroundColor": "white",
                  "borderColor": "#ddd",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "elevation": 3,
                  "margin": 10,
                  "overflow": "hidden",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 5,
                }
              }
            >
              <View>
                <View
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "height": 180,
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <FastImageView
                    defaultSource={null}
                    resizeMode="cover"
                    source={
                      {
                        "priority": "normal",
                        "uri": "https://www.top10berlin.de/wp-content/uploads/2023/11/top10berlin_feedback002.jpg",
                      }
                    }
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  />
                </View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "rgba(0, 0, 0, 0.5)",
                      "bottom": 0,
                      "height": 180,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#fff",
                        "fontSize": 20,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Closed
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "padding": 12,
                  }
                }
              >
                <View
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#333",
                        "fontSize": 18,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Lunchpod
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#777",
                        "fontSize": 14,
                        "marginTop": 4,
                      }
                    }
                  >
                    Relax your belly
                  </Text>
                </View>
                <View
                  style={
                    {
                      "backgroundColor": "#e0f2fe",
                      "borderRadius": 6,
                      "paddingHorizontal": 12,
                      "paddingVertical": 6,
                      "width": 70,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#007aff",
                        "fontWeight": "bold",
                        "textAlign": "center",
                      }
                    }
                  >
                    20-25 min
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "marginVertical": 10,
                  }
                }
              >
                <RNSVGSvgView
                  bbHeight="1"
                  bbWidth="100%"
                  focusable={false}
                  height="1"
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 1,
                        "width": "100%",
                      },
                    ]
                  }
                  width="100%"
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGLine
                      color="#000"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                          "strokeDasharray",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4291611852,
                          "type": 0,
                        }
                      }
                      strokeDasharray={
                        [
                          "4",
                          "3",
                        ]
                      }
                      strokeWidth="1"
                      x1="0"
                      x2="100%"
                      y1="1"
                      y2="1"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <View
                style={
                  {
                    "flexDirection": "row",
                    "marginBottom": 9,
                    "marginLeft": 12,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#555",
                      "fontSize": 14,
                      "marginRight": 10,
                    }
                  }
                >
                  ðŸš´ 
                  EUR
                   
                  2.15
                </Text>
                <Text
                  style={
                    {
                      "color": "#555",
                      "fontSize": 14,
                      "marginRight": 10,
                    }
                  }
                >
                  ðŸ™‚ 
                  6
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="cuisine-item"
          >
            <View
              style={
                {
                  "backgroundColor": "white",
                  "borderColor": "#ddd",
                  "borderRadius": 12,
                  "borderWidth": 1,
                  "elevation": 3,
                  "margin": 10,
                  "overflow": "hidden",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 2,
                    "width": 0,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 5,
                }
              }
            >
              <View>
                <View
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "height": 180,
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <FastImageView
                    defaultSource={null}
                    resizeMode="cover"
                    source={
                      {
                        "priority": "normal",
                        "uri": "https://www.top10berlin.de/wp-content/uploads/2023/11/umami_top10berlin003.jpg",
                      }
                    }
                    style={
                      {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  />
                </View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "rgba(0, 0, 0, 0.5)",
                      "bottom": 0,
                      "height": 180,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#fff",
                        "fontSize": 20,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Closed
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "padding": 12,
                  }
                }
              >
                <View
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#333",
                        "fontSize": 18,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Magneato
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#777",
                        "fontSize": 14,
                        "marginTop": 4,
                      }
                    }
                  >
                    Soul for food
                  </Text>
                </View>
                <View
                  style={
                    {
                      "backgroundColor": "#e0f2fe",
                      "borderRadius": 6,
                      "paddingHorizontal": 12,
                      "paddingVertical": 6,
                      "width": 70,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#007aff",
                        "fontWeight": "bold",
                        "textAlign": "center",
                      }
                    }
                  >
                    15-20 min
                  </Text>
                </View>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "marginVertical": 10,
                  }
                }
              >
                <RNSVGSvgView
                  bbHeight="1"
                  bbWidth="100%"
                  focusable={false}
                  height="1"
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 1,
                        "width": "100%",
                      },
                    ]
                  }
                  width="100%"
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGLine
                      color="#000"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                      propList={
                        [
                          "stroke",
                          "strokeWidth",
                          "strokeDasharray",
                        ]
                      }
                      stroke={
                        {
                          "payload": 4291611852,
                          "type": 0,
                        }
                      }
                      strokeDasharray={
                        [
                          "4",
                          "3",
                        ]
                      }
                      strokeWidth="1"
                      x1="0"
                      x2="100%"
                      y1="1"
                      y2="1"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <View
                style={
                  {
                    "flexDirection": "row",
                    "marginBottom": 9,
                    "marginLeft": 12,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#555",
                      "fontSize": 14,
                      "marginRight": 10,
                    }
                  }
                >
                  ðŸš´ 
                  EUR
                   
                  1.91
                </Text>
                <Text
                  style={
                    {
                      "color": "#555",
                      "fontSize": 14,
                      "marginRight": 10,
                    }
                  }
                >
                  ðŸ™‚ 
                  6
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaView>
`;
